pyam/utils.py

```diff
@@ -252,41 +252,47 @@ def _format_from_database(df):
 def _intuit_column_groups(df, index):
     """Check and categorise columns in dataframe"""
 
+    existing_cols = pd.Index(df.index.names).dropna()  # skip unnamed columns
```

Comment:
What's the use case for unnamed columns? My intuition would be that this is invalid and should raise an error, or not?

Reply:
Well, yes that could be better handled.

I added it during bugfix for the non-fast path case. Since the `reset_index` happens before calling to `_intuit_column_groups`, there will be an unnamed `RangeIndex`, ie. `df.index.names = [None]`.

In that case we just want to start with an empty index.

We could replace this by
```
existing_cols = pd.Index([]) if df.index.names == [None] else pd.Index(df.index.names)

Reply:
Ok, let's leave as is and I'll try to write a few more tests for corner cases...

Reply:
I was a bit unhappy with this and after some more thinking, i came to the conclusion it is easier to make this choice explicit by adding a `include_index` argument to `_intuit_column_groups`.

That's how it is now.